### PR TITLE
Add a config

### PR DESCRIPTION
**Pull request by km127pl**
- also adds a readme and a issue template
- as well as moves build to `dist/`

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "port": "25565"
+}
```
Comment (zefir-git) [inline config.json:3]: pls add empty line at the end here

File: .github/ISSUE_TEMPLATE [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        - Windows
+        - Linux
+        - macOS
+        - Other
+  - type: textarea
+    id: logs
+    attributes:
+      label: Relevant log output
+      description: Please copy and paste any relevant log output.
+      render: shell
```
Comment (zefir-git) [inline .github/ISSUE_TEMPLATE:56]: here too

File: src/Config.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { readFile, stat } from "node:fs/promises";
+
+export class Config {
```
Comment (zefir-git) [inline src/Config.ts:3]:
```suggestion
export default class Config {
```

File: src/Config.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+     * @returns a promise that resolves to a Config instance
+     */
+    public static async fromFile(file: string): Promise<Config> {
+        try {
+            const stats = await stat(file);
+            if (!stats.isFile()) throw new Error("Config file is not a file");
+        } catch {
+            throw new Error("Config file does not exist");
+        }
+        const data = await readFile(file, "utf-8");
```
Comment (zefir-git) [inline src/Config.ts:18]: I think here it would be better to `fs.open(file, "r")`

File: src/Server.ts [hunk 5]
```diff
@@ -1,22 +1,27 @@
 import * as net from "node:net";
 import EventEmitter from "node:events";
 import Packet from "./Packet.js";
+import path from "path";
+import { Config } from "./Config.js";
+import { PathLike } from "node:fs";
```
Comment (zefir-git) [inline src/Server.ts:6]:
```suggestion
import * as net from "node:net";
import EventEmitter from "node:events";
import { PathLike } from "node:fs";
import path from "node:path";
import Packet from "./Packet.js";
import { Config } from "./Config.js";
```

move node modules above our stuff just for better readability

File: tsconfig.json [hunk 6]
```diff
@@ -49,7 +49,7 @@
     // "emitDeclarationOnly": true,                      /* Only output d.ts files and not JavaScript files. */
     // "sourceMap": true,                                /* Create source map files for emitted JavaScript files. */
     // "outFile": "./",                                  /* Specify a file that bundles all outputs into one JavaScript file. If 'declaration' is true, also designates a file that bundles all .d.ts output. */
-    // "outDir": "./",                                   /* Specify an output folder for all emitted files. */
+    "outDir": "./dist",                                   /* Specify an output folder for all emitted files. */
     // "removeComments": true,                           /* Disable emitting comments. */
```
Comment (zefir-git) [inline tsconfig.json:53]: idk if the `/*` is aligned with the other ones but pls if its not make it xd

File: index.ts [hunk 7]
```diff
@@ -1,6 +1,9 @@
+import { Config } from "./src/Config.js";
 import Server from "./src/Server.js";
 
-const server = new Server(25566);
+const config: Config = await Config.fromFile("config.json");
```
Comment (zefir-git) [inline index.ts:4]: here it would be best imo to `path.join(Server.path, config.json)`

**Review by zefir-git — CHANGES_REQUESTED** [review]

File: index.ts (hunk 7 above)
Comment (km127pl) [reply index.ts:4]: somehow this breaks compatibility with windows, it worked on arch but does not on windows, it appends a `\` at the start for some odd reason, will just realise it as it is rn

**Review by zefir-git — DISMISSED** [review]

**Review by zefir-git — APPROVED** [review]

**Pull request merged**